contracts/GovHub.template

```diff
@@ -243,8 +244,8 @@ contract GovHub is System, IParamSubscriber {
 
   /// Execute the proposal
   /// @param proposalId The proposal Id
-  function execute(uint256 proposalId) public payable onlyInit onlyIfSuccessfulProposal(proposalId) {
-    Proposal storage proposal = proposals[proposalId];
+  function execute(uint256 proposalId) public payable nonReentrant onlyInit onlyIfSuccessfulProposal(proposalId) { //@openissue
+    Proposal storage proposal = proposals[proposalId]; //@openissue
```

Comment:
a lot of storage access going on below. consider loading proposal to memory (requires factoring out the internal mapping, see above)